### ISSUE TITLE
false positive on with `err == io.EOF` on `*bytes.Reader`

### ISSUE THREAD

**Issue by Trim21** [polyfloyd/go-errorlint#13]
this should be good:

```golang
type CompressedFile struct {
	reader             *bytes.Reader
	zipPath            string
}

func (c CompressedFile) Read(p []byte) (int, error) {
	n, err := c.reader.Read(p)
	if err == io.EOF { // report error here
		return n, io.EOF
	}

	return n, errors.Wrapf(err, "can't read from reader %s", c.zipPath)
}
```

( and yes, I know I should use `io.Reader` here instead of `*bytes.Reader`...

**Comment by polyfloyd** [comment]
Thanks for reporting!
